runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -347,38 +349,17 @@ public void subscribeListener(@Nonnull StreamListener streamListener, @Nonnull S
      * @param streamTag        only updates of tables with the stream tag will be polled
      * @param tablesOfInterest only updates from these tables of interest will be sent to listener
      */
-    public void subscribeLRCrossNamespaceListener(@Nonnull StreamListener streamListener,
+    public void subscribeLRCrossNamespaceListener(@Nonnull LRMultiNamespaceListener streamListener,
                                          @Nonnull String namespace, @Nonnull String streamTag,
                                          @Nonnull List<String> tablesOfInterest) {
-        this.subscribeLRCrossNamespaceListener(streamListener, namespace, streamTag, tablesOfInterest, null);
-    }
+        LogReplicationUtils logReplicationUtils = new LogReplicationUtils();
+        Timestamp subscriptionTimestamp = logReplicationUtils
+                .performInitialMergeAndGetSubscriptionTimestamp(streamListener, this);
```

Comment:
I believe this should run on a separate thread, otherwise you're blocking the application code until subscription is successfully completed. Take the case of, a full sync in progress, which takes a long time, the app caller attempting to subscribe could return right away and does not need to get blocked.